engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -111,11 +123,19 @@ private void createFBOs() {
         }
     }
 
-    // TODO: Pairing FBOs for swapping functionality
     public void swapReadWriteBuffers() {
-        FBO fbo = get(READONLY_GBUFFER);
-        fboLookup.put(READONLY_GBUFFER, get(WRITEONLY_GBUFFER));
-        fboLookup.put(WRITEONLY_GBUFFER, fbo);
-        notifySubscribers();
+        buffersSwapped = !buffersSwapped;
+    }
+
+    public FBO getPrimaryBuffer() {
+        return get(buffersSwapped ? BUFFER2 : BUFFER1);
+    }
+
+    public FBO getSecondaryBuffer() {
+        return get(buffersSwapped ? BUFFER1 : BUFFER2);
+    }
+
+    public FBO getFinalBuffer() {
```

Comment:
Meanwhile, I can now see one advantage of doing it this way: we don't have to pass the fboManager as input.